python/lsst/pipe/base/tests.py

```diff
@@ -0,0 +1,192 @@
+
+    Returns
+    -------
+    quantum : `lsst.daf.butler.Quantum`
+        A quantum for ``task``, when called with ``dataIds``.
+    """
+    quantum = Quantum(taskClass=type(task))
+    connections = task.config.ConnectionsClass(config=task.config)
+
+    try:
+        for name in connections.inputs:
+            connection = connections.__getattribute__(name)
+            ids = _checkDataIdMultiplicity(dataIds[name], connection.multiple)
+            for id in ids:
+                quantum.addPredictedInput(_refFromConnection(butler, connection, id))
+        for name in connections.outputs:
+            connection = connections.__getattribute__(name)
+            ids = _checkDataIdMultiplicity(dataIds[name], connection.multiple)
+            for id in ids:
+                quantum.addOutput(_refFromConnection(butler, connection, id))
+        return quantum
+    except KeyError as e:
+        raise ValueError("Mismatch in input data.") from e
+
+
+def _checkDataIdMultiplicity(dataIds, multiple):
+    if multiple:
+        if isinstance(dataIds, collections.abc.Sequence):
+            return dataIds
+        else:
+            raise ValueError(f"Expected multiple data IDs, got {dataIds}.")
+    else:
+        if isinstance(dataIds, collections.abc.Mapping) or isinstance(dataIds, DataCoordinate):
+            return [dataIds]
+        else:
+            raise ValueError(f"Expected single data ID, got {dataIds}.")
+
+
+def _refFromConnection(butler, connection, dataId, **kwargs):
+    """Create a DatasetRef for a connection in a collection.
+
+    Parameters
+    ----------
+    butler : `lsst.daf.butler.Butler`
+        The collection to point to.
+    connection : `lsst.pipe.base.connectionTypes.DimensionedConnection`
+        The connection defining the dataset type to point to.
+    dataId
+        The data ID for the dataset to point to.
+    **kwargs
+        Additional keyword arguments used to augment or construct
+        a `~lsst.daf.butler.DataCoordinate`.
+
+    Returns
+    -------
+    ref : `lsst.daf.butler.DatasetRef`
+        A reference to a dataset compatible with ``connection``, with ID
+        ``dataId``, in the collection pointed to by ``butler``.
+    """
+    universe = butler.registry.dimensions
+    dataId = DataCoordinate.standardize(dataId, **kwargs, universe=universe)
+    datasetType = connection.makeDatasetType(universe)
+    try:
+        butler.registry.getDatasetType(datasetType.name)
+    except KeyError:
+        raise ValueError(f"Invalid dataset type {connection.name}.")
+    try:
+        return DatasetRef(
+            datasetType=datasetType,
+            dataId=dataId,
+        )
+    except KeyError as e:
+        raise ValueError(f"Dataset type ({connection.name}) and ID {dataId.byName()} not compatible.") \
+            from e
+
+
+def runQuantum(task, butler, quantum, mockRun=True):
+    """Run a PipelineTask on a Quantum.
```

Comment:
Since the default behavior is to return a Mock, I think this would be clearer if it were called something more specific. Maybe `validateRunQuantum`? The name `runQuantum` is too similar to `runDataRef` or just `run` and incorrectly implies that somehow, here in these validation tests, is where the running of a PipelineTask is to occur.

Reply:
Hmm... `verifyRunQuantum` or `testRunQuantum` are also a bit misleading, because they imply the function actually does any testing (when it's really just "I want to run `runQuantum` without all the quantumgraph/workflow baggage").

`driveRunQuantum` is accurate, but I'm not sure it's clear at first glance.